cmd/libsnap-confine-private/locking.c

```diff
@@ -143,6 +161,30 @@ int sc_lock_snap(const char *snap_name)
 	return sc_lock_generic(snap_name, 0);
 }
 
+void sc_verify_snap_lock(const char *snap_name)
+{
+	int lock_fd, retval;
+
+	lock_fd = open_lock(snap_name, 0);
+	debug("trying to acquire exclusive lock over snap %s", snap_name);
+	retval = flock(lock_fd, LOCK_EX | LOCK_NB);
+	if (retval == 0) {
+		/* We managed to grab the lock, the lock was not held! */
+		flock(lock_fd, LOCK_UN);
+		close(lock_fd);
+		errno = 0;
+		die("unexpectedly managed to acquire exclusive lock over snap %s", snap_name);
+	}
+	if (retval < 0) {
```

Comment:
I'd wrap this into:
```
void sc_verify_snap_lock(const char *snap_name) {
    ...
    if (retval < 0 && errno != EWOULDBLOCK) {
       die("failed to verify exclusive lock over snap %s", snap_name);
    }

    /* We tried but failed to grab the lock because the file is already
     * locked. Good, this is what we expected. */
}
```